File: python-avd/pyavd/validation_result.py
# Copyright (c) 2023-2024 Arista Networks, Inc.
# Use of this source code is governed by the Apache License 2.0
# that can be found in the LICENSE file.
from __future__ import annotations

from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from ._errors import AvdDeprecationWarning, AvdValidationError


class ValidationResult:
    """
    Object containing result of data validation.

    Attributes:
        failed: True if data is not valid according to the schema. Otherwise False.
        validation_errors: List of AvdValidationErrors containing schema violations.
        deprecation_warnings: List of AvdDeprecationWarnings containing warning for deprecated inputs.
    """

    failed: bool
    validation_errors: list[AvdValidationError]
    deprecation_warnings: list[AvdDeprecationWarning]

    def __init__(self, failed: bool, validation_errors: list | None = None, deprecation_warnings: list | None = None) -> None:
        self.failed = failed
        self.validation_errors = validation_errors or []
        self.deprecation_warnings = deprecation_warnings or []

File: rust/validation/src/validation_result.rs
// Copyright (c) 2025 Arista Networks, Inc.
// Use of this source code is governed by the Apache License 2.0
// that can be found in the LICENSE file.

use avdschema::Dump;
use serde::Serialize;

use crate::{context::Context, feedback::Feedback};

/// Result of coercion and/or validation.
/// Holds multiple Feedback objects pointing to each Issue.
#[derive(Clone, Debug, Serialize)]
pub struct ValidationResult {
    pub violations: Vec<Feedback>,
    pub coercions: Vec<Feedback>,
}
impl From<Context<'_>> for ValidationResult {
    fn from(value: Context) -> Self {
        Self {
            coercions: value.coercions,
            violations: value.violations,
        }
    }
}
impl Dump for ValidationResult {}
